app/models/user.rb

```diff
@@ -162,8 +164,50 @@ def remember_me?
     remember_token_expires_at && remember_token_expires_at > Time.zone.now
   end
 
+  def mfa_enabled?
+    !no_mfa?
+  end
+
+  def disable_mfa!
+    no_mfa!
+    self.mfa_seed = ''
+    self.mfa_recovery_codes = []
+    self.last_otp_at = nil
+    save!(validate: false)
+  end
+
+  def enable_mfa!(seed, level)
+    self.mfa_level = level
+    self.mfa_seed = seed
+    self.mfa_recovery_codes = Array.new(10).map { SecureRandom.hex(6) }
+    save!(validate: false)
+  end
+
+  def otp_verified?(otp)
+    if no_mfa? || verify_digit_otp(otp)
```

Comment:
I don't think `no_mfa?` is needed here. As far as I can tell, `opt_verified?` is always called only after ensuring that mfa is enabled.

Reply:
Yeah. If remove `no_mfa?` here, a user who has not enabled mfa will get false for this method, seems more reasonable.